### PR TITLE
[script] run CMake jinja generation separately from slc generation

### PR DESCRIPTION
**Pull request by lmnotran**
This adds an extra step in the generation of the CMake files.
- **OLD**: .slcp ---(`slc`)---> cmake files
- **NEW**: .slcp ---(`slc`)---> yaml ---(`script/generate_cmake.py`)---> cmake files 

**This change is transparent to users. No command-line changes are needed when running `./script/build` to build apps.**

Most of the filtering logic has been moved out of `slc/exporter_templates/platform` and into `script/generate_cmake.py`. This will make debugging generation errors easier in the future.

Fixes intermittent `slc` generation failures.

Other Changes:
- Pull python requirements out of `./script/bootstrap` and into `requirements.txt`
- Remove explicit CMake version check from `./script/bootstrap`.
  - CMake already checks if its version meets the version requirement specified in the top-level [CMakeLists.txt#L33](https://github.com/openthread/ot-efr32/blob/main/CMakeLists.txt#L33)

### CODE REVIEW

**Review by suveshpratapa — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -154,6 +154,9 @@ if(OT_APP_RCP)
 elseif(OT_APP_NCP OR OT_APP_CLI)
```
[the remark this answers is not in the document]
Comment (lmnotran) [reply CMakeLists.txt:154]: It's actually for the case of the sleepy-demo apps. THE `OT_APP_*` options are specifically for adding the executable targets
| **CMake Option** | **CMake executable target** |
|------------------|-----------------------------|
| `OT_APP_RCP`      | ot-rcp                      |
| `OT_APP_NCP`       | ot-ncp-ftd, ot-ncp-mtd      |
| `OT_APP_CLI`       | ot-cli-ftd, ot-cli-mtd      |

Without the `else` case, the platform lib names don't get printed when building the `sleepy-demo` apps

[the remark this answers is not in the document]
Comment (suveshpratapa) [reply CMakeLists.txt:154]: Thanks!

**Comment by hemanth-silabs** [comment]
LGTM 👍

**Review by hemanth-silabs — APPROVED** [review]

**Comment by lmnotran** [comment]
@jwhui Can you review this please?

File: script/generate_cmake.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
```
Comment (jwhui) [inline script/generate_cmake.py:1]: nit: add copyright header?

File: slc/exporter_templates/yaml/custom.project.mak [hunk 3]
```diff
@@ -0,0 +1 @@
+# This file is intentionally blank
```
Comment (jwhui) [inline slc/exporter_templates/yaml/custom.project.mak:1]: nit: add newline?

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+cmake
+Jinja2
+GitPython
+pyyaml
```
Comment (jwhui) [inline requirements.txt:4]: nit: we generally prefer to pin to specific versions

**Comment by lmnotran** [comment]
@jwhui Just addressed your comments. Should be ready for another round of review. Thanks!

**Review by jwhui — APPROVED** [review]

**Pull request merged**